lua/treesitter-context.lua

```diff
@@ -529,10 +529,25 @@ local function highlight_contexts(bufnr, ctx_bufnr, contexts)
   end
 end
 
+--- @param win integer
 --- @param lnum integer
+--- @param relnum integer
 --- @param width integer
 --- @return string
-local function build_lno_str(lnum, width)
+local function build_lno_str(win, lnum, relnum, width)
+  if vim.wo[win].statuscolumn and vim.wo[win].statuscolumn ~= ""  then
+    vim.v.lnum = lnum
+    vim.v.relnum = relnum -- TODO: this variable is currently readonly
```

Comment:
These can both be removed, both are set accordingly depending on `v:lnum` in `nvim_eval_statusline()`. (which happens to not be read-only so we don't have to change anything there in https://github.com/neovim/neovim/pull/22859).